app/javascript/components/set-ownership-form/index.jsx

```diff
@@ -8,26 +8,58 @@ import createSchema from './ownership-form.schema';
 class SetServiceOwnershipForm extends Component {
   state = {
     initialValues: {},
-  }
+    ownerOptions: [],
+    groupOptions: [],
+  };
 
   componentDidMount() {
     const { ownershipIds } = this.props;
     this.loadInitialData(ownershipIds);
   }
 
-  loadInitialData = objectIds =>
-    http.post(`/${ManageIQ.controller}/ownership_form_fields`, { object_ids: objectIds })
-      .then(data => this.setState({ initialValues: data }))
+  loadInitialData = (ownershipIds) => {
+    /**
+     * Loads all the users with id and name into the ownerOptions state.
+     */
+    http.get('/api/users?expand=resources')
+      .then((data) => {
+        const users = [];
+        for (let i = 0; i < data.resources.length; i += 1) {
+          users.push([data.resources[i].name, data.resources[i].id]);
+        }
+        this.setState({ ownerOptions: [...users.sort()] });
+      });
+
+    /**
+     * Loads all the groups with id and description into the groupOptions state.
+     */
+    http.get('/api/groups?expand=resources')
+      .then((data) => {
+        const groups = [];
+        for (let i = 0; i < data.resources.length; i += 1) {
+          groups.push([data.resources[i].description, data.resources[i].id]);
+        }
+        this.setState({ groupOptions: [...groups.sort()] });
+      });
 
 
+    /**
+     * Loads the values which are set for the current object(s).
+     * None set results in: {user: null, group: 1}
+     * Multiple selected results in: {???}
+     */
+    http.post(`/${ManageIQ.controller}/ownership_form_fields`, { object_ids: ownershipIds })
+      .then(data => this.setState({ initialValues: data }));
+  };
+
```

Comment:
@Hyperkid123 does it make sense to have these three requests as chained promises?

Reply:
Based on the context of these calls i would use `Promise.all`  and do just one `setState` to avoid multiple re-renders.

ALso while you at it @brumik could refactor this component from class to function and use hooks (use reducer for state and use effect for life cycle methods)?

Reply:
You heard the master, please do the refactoring in a separate commit :wink: